pydantic/json_schema.py

```diff
@@ -88,7 +88,7 @@
 
 
 def update_json_schema(schema: JsonSchemaValue, updates: dict[str, Any]) -> JsonSchemaValue:
-    """Update a JSON schema by providing a dictionary of updates.
+    """Update a JSON schema in-place by providing a dictionary of updates.
```

Comment:
Just a side note - isn't it bad practice to both:
1. Modify an input argument in-place
2. Return said argument

Shouldn't we do one or the other? As in, either modify the `schema` in-place, then return `None`, or create a deepcopy of the `schema`, modify it, then return that copy?

Reply:
Obviously not related to this docs PR, but just came to my attention.

Reply:
In general yes but I think this is somehow related to some backwards compatibility thing. I don't remember why I added a function for this all that time ago lol, but I would have looked at it and been like "what is the point of this" so I feel like there was one

Reply:
Okay I dug into this a bit, and I think this function used to be used in more places (I added it over 8 months ago, well before pydantic v2 was even released), and now it isn't. I think we should open a PR to deprecate it (sadly it looks like it's technically public API.. 😩), and remove all internal usage of it.

Reply:
Cool beans, I'll open an issue.

Reply:
https://github.com/pydantic/pydantic/issues/8381